app/src/main/java/org/p2p/wallet/newsend/NewSendPresenter.kt

```diff
@@ -110,7 +115,68 @@ class NewSendPresenter(
             is CurrencyMode.Usd -> token.totalInUsd
             is CurrencyMode.Token -> token.total.scaleLong()
         } ?: return
-        view?.setMaxButtonVisibility(isVisible = inputAmount != totalAvailable.toString())
+        // TODO PWN-6092 check on max sum - min sum for creation account - 0.00089088 SOL
+        view?.setMaxButtonVisibility(isVisible = inputAmount == emptyString())
+    }
+
+    private fun validateAmounts(token: Token.Active) {
```

Comment:
how about extracting this method to a separate class SendAmountValidator? Allows to make presenter more compat and responsible for viewing logic instead of domain / validation for numbers